docs/gallery_code/general/plot_custom_aggregation.py

```diff
@@ -72,7 +72,7 @@ def main():
 
     # Make an aggregator from the user function.
     SPELL_COUNT = Aggregator(
-        "spell_count", count_spells, units_func=lambda units: 1
+        "spell_count", count_spells, units_func=lambda units, **kwargs: 1
```

Comment:
I'm concerned about this change as one could argue that it is a backwards incompatible change, and would require user to update their code, if they had created a custom aggregator

I've not yet thought of an alternative so I will give it a bit more of a think!

Reply:
Very good point. I completely missed that this not backwards-compatible.

How about a `try: ... except TypeError: ...` here:

https://github.com/schlunma/iris/blob/035224197a5550fc03db9723dc641bf40b78be5e/lib/iris/analysis/__init__.py#L631

?

This is probably not the nicest solution to this, but only requires a very small change.

Reply:
Could you use [inspect.signature](https://docs.python.org/3/library/inspect.html#introspecting-callables-with-the-signature-object) to check whether `units_func` takes `**kwargs`?

Reply:
Done. Implemented by [`inspect.getfullargspec`](https://docs.python.org/3/library/inspect.html#inspect.getfullargspec) in https://github.com/SciTools/iris/pull/5084/commits/69a854f3bb7c463cc9b647d81dcbd6294e038c3e.